sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -761,12 +761,24 @@ booleanValue
     ;
 
 interval
-    : {ansi}? INTERVAL? intervalField+
-    | {!ansi}? INTERVAL intervalField*
+    : INTERVAL (errorCapturingMultiUnitsInterval | errorCapturingUnitToUnitInterval)?
+    | {ansi}? (errorCapturingMultiUnitsInterval | errorCapturingUnitToUnitInterval)
```

Comment:
`{ansi}? (errorCapturingMultiUnitsInterval | errorCapturingUnitToUnitInterval)?` is illegal as it can match anything. Note that we need to make `(errorCapturingMultiUnitsInterval | errorCapturingUnitToUnitInterval)` optional so that we can detect `select interval` and give precise error message.